src/js/controllers/ctrl-widget-creator.js

```diff
@@ -139,24 +139,10 @@ app.controller('WidgetCreatorCtrl', function (
 
 	// Starts the Creator, sending required widget data
 	const initCreator = () => {
-		let args
 		const deferred = $q.defer()
 
-		if (inst_id != null) {
-			args = [instance.name, instance.widget, keepQSet.data, keepQSet.version, BASE_URL]
-			if (widgetType !== '.swf') {
-				args.push(MEDIA_URL)
-			} // Passing MEDIA_URL breaks the SWF, so omit it for Flash widgets! The intent is to sunset Flash support relatively soon after this code is committed.
-
-			sendToCreator('initExistingWidget', args)
-		} else {
-			args = [widget_info, BASE_URL]
-			if (widgetType !== '.swf') {
-				args.push(MEDIA_URL)
-			} //  Passing MEDIA_URL breaks the SWF, so omit it for Flash widgets! The intent is to sunset Flash support relatively soon after this code is committed.
-
-			sendToCreator('initNewWidget', args)
-		}
+		if (inst_id != null) sendToCreator('initExistingWidget', [instance.name, instance.widget, keepQSet.data, keepQSet.version,  BASE_URL, MEDIA_URL, H5P_URL])
+		else sendToCreator('initNewWidget', [widget_info, BASE_URL, MEDIA_URL, H5P_URL])
```

Comment:
This works in two lines but it's a little harder to read. More verbosity would make it clearer, but would also bloat it a little. Fine either way.

```suggestion
let creatorMethod = 'initNewWidget'
let args = [BASE_URL, MEDIA_URL, H5P_URL]
if (inst_id != null) {
    creatorMethod = 'initExistingWidget'
    args = [
        instance.name,
        instance.widget,
        keepQSet.data,
        keepQSet.version,
        ...args
    ]
} else {
    args = [widget_info, ...args]
}

sendToCreator(creatorMethod, args)
```